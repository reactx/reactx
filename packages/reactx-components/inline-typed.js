/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

// This file must have the Flow annotation.
//
// This is the Flow-typed entry point for the reconciler. It should not be
// imported directly in code. Instead, our Flow configuration uses this entry
// point for the currently checked renderer (the one you passed to `yarn flow`).

export type ButtonProps = {
  id?: string,
  title?: string,
  type?: string,
  btncolor?: string,
  disabled: boolean,
  badgeProps: {
    badge: string,
    className: string,
    badgecolor: string,
  },

  buttonProps: any,
  className?: string,
  forwardedRef: {current: any},
  renderButton?: (props: any) => void,
  renderBadge?: (props: any) => void,
  onClick: (e: MouseEventHandler<T>) => void,
};
export * from './src/ReactComponents';
