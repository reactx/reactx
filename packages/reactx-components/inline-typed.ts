/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 
 */

export type BaseColor =
  | 'none'
  | 'normal'
  | 'primary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info'
  | 'dark'
  | 'light';

export type LoadingSizeType = 'tiny' | 'small' | 'medium' | 'large' | 'extra';
