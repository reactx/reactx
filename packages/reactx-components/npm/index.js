'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./reactx-components.min.js');
} else {
  module.exports = require('./reactx-components.js');
}
