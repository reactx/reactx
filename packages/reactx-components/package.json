{
  "name": "@reactx/reactx-components",
  "description": "The standard React Components",
  "version": "1.0.20",
  "author": "Behzad Abbasi <overafaik.com>",
  "bugs": {
    "url": "https://github.com/reactx/reactx/issues"
  },
  "homepage": "https://github.com/reactx/reactx/tree/master/packages/reactx-components#readme",
  "keywords": [
    "reactx",
    "components"
  ],
  "license": "MIT",
  "files": [
    "LICENSE",
    "README.md",
    "index.js",
    "index.d.ts",
    ".npmrc",
    "reactx-components.js",
    "reactx-components.css",
    "reactx-components.min.js",
    "reactx-components.min.css",
    "src/"
  ],
  "main": "./index.ts",
  "typings": "./index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/reactx/reactx.git",
    "directory": "packages/reactx-components"
  },
  "dependencies": {
    "@ag-grid-community/core": "^26.0.0",
    "@ag-grid-community/all-modules": "^26.0.0",
    "@ag-grid-enterprise/core": "^26.0.0",
    "@ag-grid-enterprise/all-modules": "^26.0.0",
    "@ag-grid-community/react": "^26.0.0",
    "react-datepicker": "^4.2.1",
    "date-fns": "^2.24.0",
    "@popperjs/core": "^2.10.1",
    "react-transition-group": "^4.4.2",
    "classnames": "^2.3.1"
  }
}
