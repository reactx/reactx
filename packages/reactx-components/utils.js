export const cleanProps = (props) => {
  Object.keys(props).forEach(function (key) {
    if (typeof props[key] === 'undefined' || props[key] === null) {
      delete props[key];
    }
  });
};
