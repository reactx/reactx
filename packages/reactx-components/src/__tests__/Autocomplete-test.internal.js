/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';
import {Autocomplete} from '@reactx/reactx-components';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
configure({adapter: new Adapter()});

const items = [
  {id: 'foo', label: 'foo'},
  {id: 'bar', label: 'bar'},
  {id: 'baz', label: 'baz'},
];
describe('Autocomplete', () => {
  let wrapper;

  describe('while running in a browser environment', () => {
    let onChange, onSelect;

    beforeEach(() => {
      jest.resetModules();
      onChange = jest.fn();
      onSelect = jest.fn();
    });

    it('should render correctly', () => {
      wrapper = (
        <div>
          <Autocomplete
            items={items}
            showArrow={true}
            getItemValue={(item) => item.label}
            renderItem={(item, highlighted) => (
              <div
                key={item.id}
                style={{backgroundColor: highlighted ? '#eee' : 'transparent'}}>
                {item.label}
              </div>
            )}
            value={items[0].id}
            onChange={onChange}
            onSelect={onSelect}
          />
        </div>
      );
      const tree = renderer.create(wrapper).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('change', () => {
    let onChange, onSelect, container, inputWrapper, currentValue;
    beforeEach(() => {
      currentValue = items[0].id;
      onChange = jest.fn((done) => {
        currentValue = done.target.value;
      });

      onSelect = jest.fn((value) => {
        currentValue = value;
      });
      wrapper = (
        <Autocomplete
          items={items}
          getItemValue={(item) => item.label}
          renderItem={(item, highlighted) => (
            <div
              key={item.id}
              style={{backgroundColor: highlighted ? '#eee' : 'transparent'}}>
              {item.label}
            </div>
          )}
          value={currentValue}
          onChange={onChange}
          onSelect={onSelect}
        />
      );
      container = mount(wrapper);
      inputWrapper = container.find('input');
    });

    it('should be default value', () => {
      expect(inputWrapper.instance().getAttribute('aria-expanded')).toBe(
        'false',
      );
      expect(currentValue).toBe('foo');
    });

    it('is change by onSelect props', () => {
      inputWrapper.simulate('focus');
      expect(container.find('div').at(1).props().children.length).toBe(3);

      inputWrapper.simulate('keydown', {key: 'ArrowDown'});
      inputWrapper.simulate('keydown', {key: 'Enter', keyCode: 13});
      expect(onChange).toHaveBeenCalledTimes(0);
      expect(onSelect).toHaveBeenCalledTimes(1);
      expect(currentValue).toBe('bar');
    });

    it('is change by onChange props', () => {
      inputWrapper.props().onChange({
        target: {
          value: 'bar',
        },
      });
      expect(onChange).toHaveBeenCalledTimes(1);
      expect(currentValue).toBe('bar');
    });
  });
});
