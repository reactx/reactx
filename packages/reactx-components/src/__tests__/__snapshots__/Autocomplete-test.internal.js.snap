// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while running in a browser environment should render correctly 1`] = `
<div>
  <div
    style="display: inline-block;"
  >
    <input
      aria-autocomplete="list"
      aria-expanded="false"
      autocomplete="off"
      role="combobox"
      value=""
    />
  </div>
</div>
`;

exports[`while running in a browser environment should render correctly with params 1`] = `
<div>
  <div
    style="display: inline-block;"
  >
    <input
      aria-autocomplete="list"
      aria-expanded="false"
      autocomplete="off"
      role="combobox"
      value="foo"
    />
  </div>
</div>
`;
